src/pages/index.js

```diff
@@ -71,24 +121,105 @@ addCardAddButton.addEventListener("click", () => {
-  editProfileFormValidator.resetValidation(); //editProfileFormValidator might need to change back Not sure what my mistake is//
+  profileHeadingInput.value = userData.name;
+  profileDescriptionInput.value = userData.about;
+  editProfileFormValidator.resetValidation();
   updateProfilePopup.open();
 });
 
+updateAvatarButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
 // ! ||--------------------------------------------------------------------------------||
-// ! ||                                 Event Handlers                                 ||
+// ! ||                                 Functions                                 ||
 // ! ||--------------------------------------------------------------------------------||
-function handleAddCardFormSubmit(data) {
-  const card = renderCard(data);
-  cardSection.addItem(card);
+function handleValidation(form) {
+  form.enableValidation();
+}
+
+function handleAddCardFormSubmit(cardData) {
+  newCardPopup.showButtonProgress(true);
+  api
+    .addNewCard(cardData.name, cardData.link)
+    .then((res) => {
+      const card = renderCard(res);
+      cardSection.addItem(card);
+      newCardPopup.reset();
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      newCardPopup.showButtonProgress(false);
+    });
 }
 
 function handleImageClick(name, link) {
   newImagePopup.open(name, link);
 }
 
 function handleProfileSubmit(userData) {
-  userInfo.setUserInfo(userData);
+  updateProfilePopup.showButtonProgress(true);
+  api
+    .updateUserInfo(userData.name, userData.about)
+    .then((user) => {
+      userInfo.setUserInfo(user);
+      updateProfilePopup.reset();
+      updateProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      updateProfilePopup.showButtonProgress(false);
+    });
+}
+
+function handleDeleteCard(cardData) {
+  deletePopup.open();
+  deletePopup.setSubmitAction(() => {
+    deletePopup.showButtonProgress(true);
+    api
+      .deleteCard(cardData._id)
+      .then((res) => {
+        cardData.handleRemoveCard();
+        deletePopup.close();
+      })
+      .catch((err) => {
+        console.error(`Error: ${err}`);
+      })
+      .finally(() => {
+        deletePopup.showButtonProgress(false);
+      });
+  });
+}
+
+function handleLikeIcon(cardData) {
+  api
+    .setLike(cardData._id, cardData._isLiked)
+    .then((res) => {
+      cardData.handleLikeIcon(res.isLiked);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    });
+}
+
+function handleEditAvatar(data) {
+  avatarPopup.showButtonProgress(true);
+  api
+    .updateProfilePicture(data.avatar)
+    .then((user) => {
+      userInfo.setUserAvatar(user.avatar);
+      avatarPopup.reset();
+      avatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors